### PR TITLE
Hycom forcing v2 + small adds / corrections

### PR DESCRIPTION
**Pull request by mhrib**
This PR replace #224:
* Basal stress stuff deleted from this PR (similar to #222)

--o--

1) Subroutines for reading HYCOM initial SST+SSS forcing and HYCOM atmospheric forcing input in NetCDF format
2) Minor corrections in popgrid_nc in ice_grid.F90: All records are the first - unlike for binary file formats
3) Add m_per_sec as precipitation unit option
4) Add "h" (hour) as accepted unit for restart-file-dump interval
5) Add optional input argument to "ice_open" subroutine in ice_read_write.f90. This makes it ready to read binary files structured in fixed size blocks as eg. used by HYCOM.

--o--

    Developer(s): Mads Hvid Ribergaard, DMI

    Please suggest code Pull Request reviewers in the column at right.
Tony??

    Are the code changes bit for bit, different at roundoff level, or more substantial?
    Yes, BFB

    Please include the link to test results or paste the summary block from the bottom of the testing output below.

    Does this PR create or have dependencies on Icepack or any other models?

    Is the documentation being updated with this PR? (Y/N)
Yes. File updated: doc/source/user_guide/ug_case_settings.rst

    If not, does the documentation need to be updated separately at a later time? (Y/N)

Note: "Documentation" includes information on the wiki and .rst files in doc/source/,
which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.

    Other Relevant Details:

ad 1) HYCOM forcing:
Do not conflict with other stuff. But nice to have for DMI :-)

ad 2) "popgrid_nc" record-numbers in ice_grid.F90
Corrected all nrec to 1, as it is the first record that is needed for each variables: htn, hte, kmt, etc.
There is ONLY one record for each variable! This is different from "popgrid"-binary, which consists of 7 variables - ie. need a specific record number.

ad 3)+4): Simple adds

ad 5) Nice to have option, which DMI have used previous and may continue to use.

### CODE REVIEW

File: doc/source/user_guide/ug_case_settings.rst [hunk 1]
```diff
@@ -362,12 +364,15 @@ Table of namelist options
    "", "``monthly``", "monthly forcing data", ""
    "", "``ncar``", "NCAR bulk forcing data", ""
    "", "``oned``", "column forcing data", ""
+   "", "``hycom_atm``", "HYCOM atm forcing in NetCDF format", ""
```
Comment (duvivier) [inline doc/source/user_guide/ug_case_settings.rst]: I would prefer this just be called "hycom" not "hycom_atm". This is how you did it for ocn_data_type and is consistent with the other options (e.g. "ncar" or "clim").

File: doc/source/user_guide/ug_case_settings.rst [hunk 2]
```diff
@@ -362,12 +364,15 @@ Table of namelist options
...
    "``ocn_data_type``", "``default``", "constant values defined in the code", ""
    "", "``clim``", "climatological data", ""
    "", "``ncar``", "POP ocean forcing data", ""
+   "", "``hycom``", "HYCOM ocean forcing data.", "Constant initial forcing"
    "``bgc_data_type``", "``default``", "constant values defined in the code", ""
    "", "``clim``", "climatological data", ""
    "", "``near``", "POP ocean forcing data", ""
```
Comment (duvivier) [inline doc/source/user_guide/ug_case_settings.rst]: Can you also go ahead and change this typo with this PR. It should be "ncar" not "near" as the option. Thanks!

**Comment by mhrib** [comment]
Fixed:

* "hycom_atm" -> "hycom" option
* "near" -> "ncar"

Affected files:
doc/source/user_guide/ug_case_settings.rst
cicecore/cicedynB/general/ice_forcing.F90

rgs
Mads
________________________________________

**Comment by mhrib** [comment]
I have no problems leaving this out of normal tests.
However, if you prefer, I can provide pseudo hycom ocean (SST+SSS fixed fields) and hycom atm (time varying) NetCDF data interpolated to GX3?

rgs
Mads

________________________________________

**Pull request merged**